src/api.rs

```diff
@@ -27,6 +27,36 @@ pub fn test() -> &'static str {
     "Hello, test!"
 }
 
+#[put("/package/<id>", data = "<package>")]
+pub async fn package_update(
+    id: String,
+    package: Json<Package>,
+    mod_db: &State<ModuleDB>,
+) -> (Status, &'static str) {
+    let mod_r = mod_db.read().await;
+    let res = mod_r.get(&id);
+    if res.is_none() {
+        return (Status::NotFound, "Package does not exist.");
+    }
+
+    let db = res.unwrap();
+
+    //update the package if metadata matches
+    if (package.metadata.Name == db.name)
+        & (package.metadata.Version == db.ver)
+        & (package.metadata.ID == db.id)
```

Comment:
Even though `&` technically works, please use `&&` for my sanity.

Also, you'll want a `else` for when they don't match up, perhaps returning 404 or 400.